File: src/main/java/rosatech/mixin/late/LateMixinLoader.java
package rosatech.mixin.late;

import rosatech.api.util.RosaMods;
import zone.rong.mixinbooter.ILateMixinLoader;

import java.util.Arrays;
import java.util.List;

@SuppressWarnings("unused")
public class LateMixinLoader implements ILateMixinLoader {

    @Override
    public List<String> getMixinConfigs() {
        String[] configs = {"mixin.rosatech.botania.json"};
        return Arrays.asList(configs);
    }

    @Override
    public boolean shouldMixinConfigQueue(String mixinConfig) {
        return switch (mixinConfig) {
            case "mixin.rosatech.botania.json" -> RosaMods.Botania.isModLoaded();
            default -> ILateMixinLoader.super.shouldMixinConfigQueue(mixinConfig);
        };
    }
}

File: src/main/kotlin/rosatech/mixin/late/LateMixinLoader.kt
package rosatech.mixin.late

import net.minecraftforge.fml.common.Loader
import zone.rong.mixinbooter.ILateMixinLoader
import java.util.*

@Suppress("unused")
class LateMixinLoader : ILateMixinLoader {
    override fun getMixinConfigs(): List<String> {
        val configs = arrayOf<String>(
            "mixins.rosatech.late.json",
            "mixins.rosatech.late.thaumcraft.json"
        )
        return Arrays.asList(*configs)
    }

    override fun shouldMixinConfigQueue(mixinConfig: String): Boolean {
        return when (mixinConfig) {
            "mixins.rosatech.late.thaumcraft.json" -> Loader.isModLoaded("thaumcraft")
            else -> super.shouldMixinConfigQueue(mixinConfig)
        }
    }
}
